pkg/compat/client.go

```diff
@@ -219,3 +222,31 @@ func (c client) Update(ctx context.Context, in runtime.Object) error {
 	}
 	return c.Client.Update(ctx, obj)
 }
+
+//
+// Parses gitVersion ("v1.11.0+d4cacc0"), returns major, minor, err
+func parseGitVersion(gitVersion string) (int, int, error) {
+	r, _ := regexp.Compile(`v[0-9]+\.[0-9]+`)
+	valid := r.MatchString(gitVersion)
+	if !valid {
+		return 0, 0, errors.New("gitVersion does not match expected format")
+	}
+	versionArr := strings.Split(gitVersion, "v")
+	if len(versionArr) < 2 {
+		return 0, 0, errors.New("gitVersion missing leading 'v'")
+	}
+	majorMinorArr := strings.Split(versionArr[1], ".")
+	if len(majorMinorArr) < 2 {
+		return 0, 0, errors.New("gitVersion major + minor parse failed")
+	}
```

Comment:
Won't the `MatchString()` catch the missing `v` and `.` errors?
I don't see how these checks would be reached.

Reply:
@jortel I would think yes, but was trying to be extra safe to avoid controller panic if I try to index into something that doesn't exist.

Reply:
I thought of a way to make this a little bit cleaner.